src/main/java/io/cdap/plugin/fileappend/FileAppenderSink.java

```diff
@@ -53,131 +51,76 @@
   "than configurable thresholds.")
 public class FileAppenderSink extends BatchSink<StructuredRecord, NullWritable, Text> {
   public static final String NAME = "FileAppender";
-  private final Conf config;
+  private final FileAppenderSinkConfig config;
   private Schema outputSchema;
 
-  /**
-   * Config properties for the plugin.
-   */
-  public static class Conf extends PluginConfig {
-
-    @Description("The name of the FileSet to write to.")
-    private String name;
-
-    @Nullable
-    @Description("Optional directory within the FileSet to write to.")
-    private String outputDir;
-
-    @Description("The prefix to use for all files in the FileSet.")
-    private String fileNamePrefix;
-
-    @Nullable
-    @Description("Rotate files after they are this large (in mb). 0 means no limit. Defaults to 100.")
-    private Long sizeThreshold;
-
-    @Nullable
-    @Description("Rotate files after they are this old (in minutes). 0 means no limit. Defaults to 60.")
-    private Long ageThreshold;
-
-    @Nullable
-    @Description("The separator to use to join input record fields together. Defaults to ','.")
-    private String fieldSeparator;
-
-    @Nullable
-    @Description("The schema of the FileSet.")
-    private String schema;
-
-    public Conf() {
-      fieldSeparator = ",";
-      sizeThreshold = 100L;
-      ageThreshold = 60L;
-    }
-
-    public void validate() {
-      if (sizeThreshold < 0) {
-        throw new IllegalArgumentException("sizeThreshold must be at least 0.");
-      }
-      if (ageThreshold < 0) {
-        throw new IllegalArgumentException("ageThreshold must be at least 0.");
-      }
-    }
-  }
-
-  public FileAppenderSink(Conf config) {
+  public FileAppenderSink(FileAppenderSinkConfig config) {
     this.config = config;
   }
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    config.validate();
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(failureCollector);
     // TODO: validate input schema
```

Comment:
lets add validation for input schema as well.

Reply:
I'm not sure what should be validated in input schema. Do you have any requirements? As I can see, existing validation may be enough in most cases.